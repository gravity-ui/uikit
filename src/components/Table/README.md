<!--GITHUB_BLOCK-->

## Table

<!--/GITHUB_BLOCK-->

```jsx
import {Table} from '@gravity-ui/uikit';
```

A table that allows the selecting and sorting of rows, and performing actions on a row.

<!--GITHUB_BLOCK-->

Additional functionality is enabled via HOCs:

- [withTableActions](#usage-with-hoc-withtableactions)
- [withTableCopy](#usage-with-hoc-withtablecopy)
- [withTableSelection](#usage-with-hoc-withtableselection)
- [withTableSettings](#usage-with-hoc-withtablesettings)
- [withTableSorting](#usage-with-hoc-withtablesorting)

<!--/GITHUB_BLOCK-->

## Properties

| Name                             | Description                                                                                                                                                                     |                                        Type                                        |   Default   |
| :------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | :--------------------------------------------------------------------------------: | :---------: |
| data                             | Data                                                                                                                                                                            |                                      `any[]`                                       |             |
| columns                          | Column parameters                                                                                                                                                               |                               `TableColumnConfig[]`                                |             |
| verticalAlign                    | Vertical alignment of contents                                                                                                                                                  |                                 `"top"` `"middle"`                                 |             |
| getRowDescriptor                 | Handler to get row descriptor                                                                                                                                                   |                   `(item: any, index: number) => DescriptorType`                   |             |
| getRowId                         | The row ID, used when selecting and sorting rows. If you skip a row, its ID will be the value of the field in the row data with the same name as the column ID                  |                 `string` `((item: any, index: number) => string)`                  |             |
| getRowClassNames                 | Row CSS classes                                                                                                                                                                 |                      `(item: any, index: number) => string[]`                      |             |
| isRowDisabled                    | Condition for disabling columns                                                                                                                                                 |                      `(item: any, index: number) => boolean`                       |             |
| onRowClick                       | Row click handler                                                                                                                                                               | `(item: any, index: number, event: React.MouseEvent<HTMLTableRowElement>) => void` |             |
| onRowMouseEnter                  | Row mouseenter handler                                                                                                                                                          | `(item: any, index: number, event: React.MouseEvent<HTMLTableRowElement>) => void` |             |
| onRowMouseLeave                  | Row mouseleave handler                                                                                                                                                          | `(item: any, index: number, event: React.MouseEvent<HTMLTableRowElement>) => void` |             |
| emptyMessage                     | The message returned if data is missing.                                                                                                                                        |                                      `string`                                      | `"No data"` |
| className                        | Table CSS class                                                                                                                                                                 |                                      `string`                                      |             |
| edgePadding                      | Adds horizontal padding for edge cells                                                                                                                                          |                                     `boolean`                                      |             |
| stickyHorizontalScroll           | A horizontal sticky scroll in a table. NB: A table cannot have a fixed height and a sticky scroll at the same time. A sticky scroll will not work if the table has an overflow. |                                     `boolean`                                      |   `false`   |
| stickyHorizontalScrollBreakpoint | The threshold that the parent block should reach before making a scroll sticky. This is useful in the console, for example, when the groupActions bar closes the scroll.        |                                      `number`                                      |     `0`     |

### DescriptorType

| Name       | Description                                      |    Type     | Default |
| :--------- | :----------------------------------------------- | :---------: | :-----: |
| id         | The row ID, used when selecting and sorting rows |  `string`   |         |
| disabled   | Condition for disabling columns                  |  `boolean`  |         |
| classNames | Row CSS classes                                  | ` string[]` |         |

### TableColumnConfig

| Name        | Description                                                                                                        |                            Type                            |                           Default                           |
| :---------- | :----------------------------------------------------------------------------------------------------------------- | :--------------------------------------------------------: | :---------------------------------------------------------: |
| id          | Column ID                                                                                                          |                          `string`                          |                                                             |
| name        | Column name (header)                                                                                               |             `string` `(() => React.ReactNode)`             |                          column ID                          |
| className   | CSS-class that will be added to all cells in the column                                                            |                          `string`                          |                                                             |
| placeholder | The stub when there is no data in a cell                                                                           | `string` `((item: any, index: number) => React.ReactNode)` |                        `â€” (&mdash;)`                        |
| template    | Cell contents. If you skip a row, the cell contents will be the value of the field with the same name as this row. | `string` `((item: any, index: number) => React.ReactNode)` | The value of the field with the name equal to the column ID |
| align       | Content alignment                                                                                                  |                `"start"` `"center"` `"end"`                |                                                             |
| sticky      | Sticky column                                                                                                      |                     `"start"` `"end"`                      |                                                             |
| primary     | Distinguishes a column among other                                                                                 |                         `boolean`                          |                                                             |
| width       | Column's content width in px                                                                                       |                     `number` `string`                      |                                                             |
| meta        | Various data, HOC settings                                                                                         |                   `Record<string, any>`                    |                                                             |

## Usage with HOC `withTableActions`

Adds a special column with actions to table columns.

### Properties

| Name           | Description                                 |                        Type                         |
| :------------- | :------------------------------------------ | :-------------------------------------------------: |
| getRowActions  | Array of action configs for each row        | `(item: any, index: number) => TableActionConfig[]` |
| rowActionsSize | Size of actions button and popup menu items |              `"s"` `"m"` `"l"` `"xl"`               |

### TableActionConfig

```ts
type TableActionConfig = TableAction | TableActionGroup;
```

#### TableAction

| Name     | Description                                                        |                 Type                 |  Default   |
| :------- | :----------------------------------------------------------------- | :----------------------------------: | :--------: |
| text     | Text                                                               |               `string`               |            |
| handler  | Click handler                                                      | `(item: any, index: number) => void` |            |
| disabled | Action disabled                                                    |              `boolean`               |            |
| href     | Menu item with this prop becomes a link to the specified location. |               `string`               |            |
| target   | Same as the `target` attribute of the `<a>` tag.                   |               `string`               |            |
| rel      | Same as the `rel` attribute of the `<a>` tag.                      |               `string`               |            |
| theme    | Theme                                                              |        `"normal"` `"danger"`         | `"normal"` |

#### TableActionGroup

| Name  | Description         |         Type          |
| :---- | :------------------ | :-------------------: |
| title | Action group header |       `string`        |
| items | Action group items  | `TableActionConfig[]` |

### Example

```jsx
import {Table, withTableActions} from '@gravity-ui/uikit';

const MyTable = withTableActions(Table);
const data = [
  {id: 1, text: 'Hello'},
  {id: 2, text: 'World'},
];
const columns = [{id: 'id'}, {id: 'text'}];
const getRowActions = () => {
  return [
    {
      text: 'Print',
      handler: () => {},
    },
    {
      text: 'Remove',
      handler: () => {},
      theme: 'danger',
    },
  ];
};

const table = <MyTable data={data} columns={columns} getRowActions={getRowActions} />;
```

## Usage with HOC `withTableCopy`

Allows the contents of a cell or any text to be copied.

### ColumnMeta

| Name | Description                                                             |                                            Type                                             |
| :--- | :---------------------------------------------------------------------- | :-----------------------------------------------------------------------------------------: |
| copy | The text to be copies. If true is passed, the cell contents are copied. | `boolean` `((item: any, index: number) => string)` `((item: any, index: number) => number)` |

### Example

```jsx
import {Table, withTableCopy} from '@gravity-ui/uikit';

const MyTable = withTableCopy(Table);
const data = [
  {id: 1, text: 'Hello'},
  {id: 2, text: 'World'},
];
const columns = [
  {id: 'id', meta: {copy: ({id}) => `ID #${id}`}},
  {id: 'text', meta: {copy: true}},
];

const table = <MyTable data={data} columns={columns} />;
```

## Usage with HOC `withTableSelection`

Enables the selection of table rows.

### Properties

| Name              | Description                 |           Type            |
| :---------------- | :-------------------------- | :-----------------------: |
| selectedIds       | Rows selected               |        `string[]`         |
| onSelectionChange | Selected row change handler | `(ids: string[]) => void` |

### Example

```jsx
import {Table, withTableSelection} from '@gravity-ui/uikit';

const MyTable = withTableSelection(Table);
const data = [
  {id: 1, text: 'Hello'},
  {id: 2, text: 'World'},
];
const columns = [{id: 'id'}, {id: 'text'}];
const getRowId = 'id';

function SelectionTable() {
  const [selectedIds, setSelectedIds] = React.useState([1]);

  return (
    <MyTable
      data={data}
      columns={columns}
      getRowId={getRowId}
      selectedIds={selectedIds}
      onSelectionChange={setSelectedIds}
    />
  );
}
```

## Usage with HOC `withTableSettings`

Enables functionality for table column settings. You can use ut in two forms:

```jsx
import {Table, withTableSettings} from './withTableSettings';

// No options passed
const MyTable1 = withTableSettings(Table);
// or with options
const MyTable1 = withTableSettings({sortable: false})(Table);
```

### Options

| Name     | Description                                       |      Type      | Default |
| :------- | :------------------------------------------------ | :------------: | :-----: |
| width    | Settings' popup width                             | `number` `fit` |         |
| sortable | Whether or not add ability to sort settings items |   `boolean`    | `true`  |

### ColumnMeta

| Name              | Description                                                                |   Type    | Default |
| :---------------- | :------------------------------------------------------------------------- | :-------: | :-----: |
| selectedByDefault | Specifies whether a column is selected if it is missing from the settings. | `boolean` | `true`  |
| selectedAlways    | Makes the column always selected. You cannot change its visibility.        | `boolean` | `false` |

### Properties

| Name               | Description                   |                     Type                     |
| :----------------- | :---------------------------- | :------------------------------------------: |
| settingsPopupWidth | TableColumnSetup pop-up width |                `number` `fit`                |
| settings           | Current settings              |             `TableSettingsData`              |
| updateSettings     | Settings update handle        | `(data: TableSettingsData) => Promise<void>` |

### TableSettingsData

```ts
type TableSettingsData = Array<{
  id: string;
  isSelected?: boolean;
}>;
```

### Example

```jsx
import {Table, withTableSettings} from '@gravity-ui/uikit';

const MyTable = withTableSettings({width: 100, sortable: false})(Table);
const data = [
  {id: 1, text: 'Hello'},
  {id: 2, text: 'World'},
];
const columns = [{id: 'id'}, {id: 'text'}];

function SelectionTable() {
  const initialSettings = [
    {id: 'id', isSelected: false},
    {id: 'text', isSelected: true},
  ];
  const [settings, setSettings] = React.useState(initialSettings);

  return (
    <MyTable
      data={data}
      columns={columns}
      settings={settings}
      updateSettings={(settings) => {
        setSettings(settings);
        return Promise.resolve();
      }}
    />
  );
}
```

## Usage with HOC `withTableSorting`

Enables column sorting.

### ColumnMeta

| Name             | Description                                                                                                                                               |                       Type                       | Default |
| :--------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------------: | :-----: |
| defaultSortOrder | Sets the primary sorting order                                                                                                                            |                 `"asc"` `"desc"`                 |  `asc`  |
| sort             | The sorting function. It should return a value for sorting in ascending order. If true is passed, cell values are compared and sorted in ascending order. | `boolean` `((itemA: any, itemB: any) => number)` |         |

### Properties

| Name              | Description                                         |                 Type                  |
| :---------------- | :-------------------------------------------------- | :-----------------------------------: |
| defaultSortState  | Default sorting state for an uncontrolled component |           `TableSortState`            |
| sortState         | Sorting state                                       |           `TableSortState`            |
| onSortStateChange | Sorting state change handle                         | `(sortState: TableSortState) => void` |

If the `sortState` and `onSortStateChange` props are not passed, the sorting state is stored in the component itself.

### TableSortState

```ts
type TableSortState = Array<{
  column: string;
  order: 'asc' | 'desc';
}>;
```

### Example

```jsx
import {Table, withTableSorting} from '@gravity-ui/uikit';

const MyTable = withTableSorting(Table);
const data = [
  {id: 1, text: 'Hello', date: '2016-10-25'},
  {id: 2, text: 'World', date: '2020-08-15'},
];
const columns = [
  {id: 'id', meta: {sort: true}},
  {
    id: 'text',
    meta: {defaultSortOrder: 'desc', sort: (a, b) => Date.parse(a.date) - Date.parse(b.date)},
  },
];

const table = <MyTable data={data} columns={columns} />;
```
