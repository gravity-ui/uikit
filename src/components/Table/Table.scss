@use '../../../styles/mixins';
@use './variables';

#{variables.$block} {
    overflow: auto;
    position: relative;

    &__scroll-container {
        overflow: auto;
        position: relative;

        -ms-overflow-style: none;
        scrollbar-width: none;

        &::-webkit-scrollbar {
            display: none;
        }
    }

    &__horizontal-scroll-bar {
        overflow-x: auto;
        margin-top: -1px;

        &-inner {
            height: 1px;
            position: relative;

            // Safari 13 does not support transparent elements
            &::before {
                content: '';
                position: absolute;
                width: 1px;
                height: 1px;
                top: 0;
                left: 0;
                background-color: rgba(255, 255, 255, 0.01);
            }
        }

        &_sticky-horizontal-scroll {
            position: sticky;
            z-index: 3;
        }
    }

    &__table {
        border-spacing: 0;
        // fix border disappear in Firefox:
        border-collapse: separate;
    }

    &__cell {
        @include mixins.overflow-ellipsis();

        position: relative;
        box-sizing: content-box;
        padding: 11px 10px;
        border-bottom: 1px solid var(--g-color-line-generic);

        line-height: variables.$tableLineHeight;
        text-align: left;
        overflow-wrap: break-word;

        &:first-child {
            padding-left: 0;
        }

        &:last-child {
            padding-right: 0;
        }

        &:not(&_word-wrap) {
            @include mixins.overflow-ellipsis();
        }

        &_align_center {
            text-align: center;
        }

        &_align_right {
            text-align: right;
        }

        &_sticky_left,
        &_sticky_right {
            position: sticky;
            z-index: 2;
        }

        &_sticky_left,
        &_sticky_right {
            background: var(--g-color-base-background);
        }

        &_border_right {
            border-right: 1px solid var(--g-color-line-generic);
        }

        &_edge-padding {
            &:first-child {
                padding-left: 15px;
            }

            &:last-child {
                padding-right: 15px;
            }
        }
    }

    &__row {
        &_vertical-align_top {
            vertical-align: top;
        }

        &_vertical-align_middle {
            vertical-align: middle;
        }
    }

    &__row_empty &__cell {
        text-align: center;
    }

    &__body &__row:last-child &__cell {
        border-bottom-color: transparent;
    }

    &__th-content {
        @include mixins.text-accent;
        display: inline-block;
        &::first-letter {
            text-transform: uppercase;
        }
    }

    &__body &__row_interactive:hover {
        background-color: var(--g-color-base-simple-hover-solid);
        cursor: pointer;

        #{variables.$block}__cell_sticky_left,
        #{variables.$block}__cell_sticky_right {
            background: var(--g-color-base-simple-hover-solid);
        }
    }

    &__body &__row_disabled {
        opacity: 0.3;
    }

    &_with-primary &__body &__cell {
        color: var(--g-color-text-secondary);

        &_primary {
            color: var(--g-color-text-primary);
        }
    }

    // forcing overflow:visible so we are not breaking sticky scroll
    // (because sticky scroll relates to closest parent with overflow)
    &_with-sticky-scroll {
        overflow: visible;
    }
}
