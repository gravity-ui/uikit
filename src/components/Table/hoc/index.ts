export {withTableSelection} from './withTableSelection/withTableSelection';
export type {WithTableSelectionProps} from './withTableSelection/withTableSelection';
export {withTableActions} from './withTableActions/withTableActions';
export type {
    RenderRowActionsProps,
    WithTableActionsProps,
    TableActionConfig,
    TableAction,
    TableActionGroup,
} from './withTableActions/withTableActions';
export {withTableCopy} from './withTableCopy/withTableCopy';
export type {WithTableCopyProps} from './withTableCopy/withTableCopy';
export {withTableSorting, TableSortIndicator} from './withTableSorting/withTableSorting';
export type {
    WithTableSortingProps,
    TableSortState,
    TableColumnSortState,
} from './withTableSorting/withTableSorting';
export {withTableSettings} from './withTableSettings/withTableSettings';
export type {
    WithTableSettingsProps,
    TableSettingsData,
    TableColumnSetupItem,
} from './withTableSettings/withTableSettings';
export * from './withTableSettings/TableColumnSetup/TableColumnSetup';
