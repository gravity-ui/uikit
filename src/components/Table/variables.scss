@use '../variables';

$block: '.#{variables.$ns}table';
$tableLineHeight: 18px;
