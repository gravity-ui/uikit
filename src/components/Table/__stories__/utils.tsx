import {withTableSelection} from '..';
import {Table} from '../Table';
import type {TableColumnConfig} from '../Table';
import {withTableActions, withTableCopy, withTableSettings, withTableSorting} from '../hoc';

export interface DataItem {
    name: string;
    city?: string;
    phone: string;
    count: number;
    date: string;
    disabled?: boolean;
}

export const data: DataItem[] = [
    {
        name: 'Nomlanga Compton',
        city: 'Erli',
        phone: '+7 (923) 737-89-72',
        count: 82,
        date: '2019-03-15',
    },
    {
        name: 'Paul Hatfield',
        city: 'Campitello di Fassa',
        phone: '+7 (900) 333-82-02',
        count: 51,
        date: '2019-11-23',
    },
    {
        name: 'Phelan Daniel',
        city: 'Meugliano',
        phone: '+7 (925) 549-50-23',
        count: 10,
        date: '2019-05-14',
    },
    {
        name: 'Hiram Mayer',
        city: '',
        phone: '+7 (950) 372-56-84',
        count: 54,
        date: '2019-03-29',
    },
    {
        name: 'Madeline Puckett',
        phone: '+7 (908) 582-05-91',
        count: 75,
        date: '2019-02-01',
        disabled: true,
    },
];

export const columns: TableColumnConfig<DataItem>[] = [
    {
        id: 'name',
        name: 'Name',
    },
    {
        id: 'city',
        name: 'City',
    },
    {
        id: 'phone',
        name: 'Phone',
    },
    {
        id: 'count',
        name: 'Count',
        align: 'right',
    },
    {
        id: 'date',
        name: 'Date created',
    },
];

export const TableWithAction = withTableActions<DataItem>(Table);
export const TableWithCopy = withTableCopy<DataItem>(Table);
export const TableWithSelection = withTableSelection<DataItem>(Table);
export const TableWithSettings = withTableSettings<DataItem>(Table);
export const TableWithSorting = withTableSorting<DataItem>(Table);
