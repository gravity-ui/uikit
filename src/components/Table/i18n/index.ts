import {addComponentKeysets} from '../../utils/addComponentKeysets';

import en from './en.json';
import ru from './ru.json';

const COMPONENT = 'Table';

export default addComponentKeysets({en, ru}, COMPONENT);
