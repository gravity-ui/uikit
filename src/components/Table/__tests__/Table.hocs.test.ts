import React from 'react';

import userEvent from '@testing-library/user-event';

import {render, screen, within} from '../../../../test-utils/utils';
import type {TableProps} from '../Table';
import {Table} from '../Table';
import type {
    WithTableActionsProps,
    WithTableSelectionProps,
    WithTableSettingsProps,
    WithTableSortingProps,
} from '../hoc';
import {
    withTableActions,
    withTableCopy,
    withTableSelection,
    withTableSettings,
    withTableSorting,
} from '../hoc';

interface Model {
    disabled: boolean;
}

function getTextContent(html = '') {
    return html.replace(/uniq\d+/g, '');
}

describe('Table HOCs tests', () => {
    it('using withTableActions and withTableSelection should not depend of order', () => {
        const Table1 = withTableActions(withTableSelection<Model>(Table));
        const Table2 = withTableSelection(withTableActions<Model>(Table));

        type Props = TableProps<Model> &
            WithTableActionsProps<Model> &
            WithTableSelectionProps<Model>;
        const props: Props = {
            data: [{disabled: false}, {disabled: true}],
            columns: [{id: 'name'}],
            isRowDisabled: ({disabled}) => disabled,
            selectedIds: [],
            onSelectionChange: () => {},
            getRowActions: () => [],
        };
        const {container: container1} = render(React.createElement<Props>(Table1, props));
        const {container: container2} = render(React.createElement<Props>(Table2, props));

        expect(getTextContent(container1.outerHTML)).toEqual(getTextContent(container2.outerHTML));
    });

    it('using withTableActions with onRowClick should not click on row when clicking on menu item', async () => {
        const TableWithActions = withTableActions<Model>(Table);

        type Props = TableProps<Model> & WithTableActionsProps<Model>;
        const user = userEvent.setup();
        const onRowClick = jest.fn();
        const onMenuItemClick = jest.fn();
        const props: Props = {
            data: [{disabled: false}],
            columns: [{id: 'name'}],
            onRowClick,
            getRowActions: () => [
                {
                    text: 'event',
                    handler: onMenuItemClick,
                },
            ],
        };

        render(React.createElement<Props>(TableWithActions, props));

        const table = screen.getByRole('table');
        const menuButton = within(table).getAllByRole('button');
        await user.click(menuButton[0]);
        const menuItem = screen.getByRole('menuitem');
        expect(menuItem).toBeInTheDocument();
        await user.click(menuItem);

        expect(onMenuItemClick).toHaveBeenCalledTimes(props.data.length);
        expect(onRowClick).toHaveBeenCalledTimes(0);
    });

    it('using withTableActions and withTableSorting should not depend of order', () => {
        const Table1 = withTableActions(withTableSorting<Model>(Table));
        const Table2 = withTableSorting(withTableActions<Model>(Table));

        type Props = TableProps<Model> & WithTableActionsProps<Model> & WithTableSortingProps;
        const props: Props = {
            data: [{disabled: false}, {disabled: true}],
            columns: [{id: 'name'}],
            isRowDisabled: ({disabled}) => disabled,
            getRowActions: () => [],
        };
        const {container: container1} = render(React.createElement<Props>(Table1, props));
        const {container: container2} = render(React.createElement<Props>(Table2, props));

        expect(getTextContent(container1.outerHTML)).toEqual(getTextContent(container2.outerHTML));
    });

    it('using all HOCs should not depend of order', () => {
        const Table1 = withTableSorting(
            withTableSettings(
                withTableCopy(withTableActions(withTableSelection<Model, {}>(Table))),
            ),
        );
        const Table2 = withTableSelection(
            withTableActions(withTableCopy(withTableSettings(withTableSorting<Model, {}>(Table)))),
        );

        type Props = TableProps<Model> &
            WithTableActionsProps<Model> &
            WithTableSelectionProps<Model> &
            WithTableSettingsProps &
            WithTableSortingProps;

        const props: Props = {
            data: [{disabled: false}, {disabled: true}],
            columns: [{id: 'name'}],
            isRowDisabled: ({disabled}) => disabled,
            selectedIds: [],
            onSelectionChange: () => {},
            getRowActions: () => [],
            updateSettings: () => Promise.resolve(),
            settings: [],
        };
        const {container: container1} = render(React.createElement<Props>(Table1, props));
        const {container: container2} = render(React.createElement<Props>(Table2, props));

        expect(getTextContent(container1.outerHTML)).toEqual(getTextContent(container2.outerHTML));
    });
});
