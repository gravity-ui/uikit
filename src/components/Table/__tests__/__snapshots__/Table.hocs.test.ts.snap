// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table HOCs tests all HOCs should match snapshot 1`] = `
<div>
  <div
    class="yc-table"
  >
    <table
      class="yc-table__table"
    >
      <thead
        class="yc-table__head"
      >
        <tr
          class="yc-table__row"
        >
          <th
            class="yc-table__cell yc-table__cell_edge-padding yc-table__checkbox_cell"
          >
            <label
              class="yc-control-label yc-control-label_size_l yc-checkbox yc-checkbox_size_l yc-table__selection-checkbox"
            >
              <span
                class="yc-control-label__indicator yc-checkbox__indicator"
              >
                <span
                  aria-hidden="true"
                  class="yc-checkbox__icon"
                >
                  <svg
                    class="yc-checkbox__icon-svg yc-checkbox__icon-svg_type_tick"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 12 10"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.49 5.385l1.644-1.644 4.385 4.385L4.874 9.77.49 5.385zm4.384 1.096L10.356 1 12 2.644 6.519 8.126 4.874 6.48v.001z"
                    />
                  </svg>
                </span>
                <input
                  aria-checked="false"
                  class="yc-checkbox__control"
                  type="checkbox"
                  value=""
                />
                <span
                  class="yc-checkbox__outline"
                />
              </span>
            </label>
          </th>
          <th
            class="yc-table__cell yc-table__cell_edge-padding"
          >
            name
          </th>
          <th
            class="yc-table__cell yc-table__cell_sticky_right yc-table__cell_edge-padding"
            style="width: 28px; right: 0px;"
          >
            <div
              class="yc-table__settings"
            >
              <div
                class="yc-table-column-setup"
              >
                <div
                  class="yc-table-column-setup__control"
                >
                  <button
                    aria-label="Table settings"
                    class="yc-button yc-button_view_flat yc-button_size_m yc-button_pin_round-round yc-table__settings-button"
                    type="button"
                  >
                    <span
                      class="yc-button__icon"
                    >
                      <span
                        class="yc-button__icon-inner"
                      >
                        <svg
                          aria-hidden="true"
                          class="yc-icon"
                          fill="currentColor"
                          height="16"
                          stroke="none"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                          xmlns:xlink="http://www.w3.org/1999/xlink"
                        >
                          <svg
                            fill="none"
                            viewBox="0 0 16 16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              clip-path="url(#a)"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M7.199 2H8.8a.2.2 0 0 1 .2.2c0 1.808 1.958 2.939 3.524 2.034a.199.199 0 0 1 .271.073l.802 1.388a.199.199 0 0 1-.073.272c-1.566.904-1.566 3.164 0 4.069a.199.199 0 0 1 .073.271l-.802 1.388a.199.199 0 0 1-.271.073C10.958 10.863 9 11.993 9 13.8a.2.2 0 0 1-.199.2H7.2a.199.199 0 0 1-.2-.2c0-1.808-1.958-2.938-3.524-2.034a.199.199 0 0 1-.272-.073l-.8-1.388a.199.199 0 0 1 .072-.271c1.566-.905 1.566-3.165 0-4.07a.199.199 0 0 1-.073-.271l.801-1.388a.199.199 0 0 1 .272-.073C5.042 5.138 7 4.007 7 2.2c0-.11.089-.199.199-.199ZM5.5 2.2c0-.94.76-1.7 1.699-1.7H8.8c.94 0 1.7.76 1.7 1.7a.85.85 0 0 0 1.274.735 1.699 1.699 0 0 1 2.32.622l.802 1.388c.469.813.19 1.851-.622 2.32a.85.85 0 0 0 0 1.472 1.7 1.7 0 0 1 .622 2.32l-.802 1.388a1.699 1.699 0 0 1-2.32.622.85.85 0 0 0-1.274.735c0 .939-.76 1.7-1.699 1.7H7.2a1.7 1.7 0 0 1-1.699-1.7.85.85 0 0 0-1.274-.735 1.698 1.698 0 0 1-2.32-.622l-.802-1.388a1.699 1.699 0 0 1 .622-2.32.85.85 0 0 0 0-1.471 1.699 1.699 0 0 1-.622-2.321l.801-1.388a1.699 1.699 0 0 1 2.32-.622A.85.85 0 0 0 5.5 2.2Zm4 5.8a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                                fill="currentColor"
                                fill-rule="evenodd"
                              />
                            </g>
                            <defs>
                              <clippath
                                id="a"
                              >
                                <path
                                  d="M0 0h16v16H0z"
                                  fill="currentColor"
                                />
                              </clippath>
                            </defs>
                          </svg>
                        </svg>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="yc-table__body"
      >
        <tr
          class="yc-table__row"
        >
          <td
            class="yc-table__cell yc-table__cell_edge-padding yc-table__checkbox_cell"
          >
            <label
              class="yc-control-label yc-control-label_size_l yc-checkbox yc-checkbox_size_l yc-table__selection-checkbox"
            >
              <span
                class="yc-control-label__indicator yc-checkbox__indicator"
              >
                <span
                  aria-hidden="true"
                  class="yc-checkbox__icon"
                >
                  <svg
                    class="yc-checkbox__icon-svg yc-checkbox__icon-svg_type_tick"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 12 10"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.49 5.385l1.644-1.644 4.385 4.385L4.874 9.77.49 5.385zm4.384 1.096L10.356 1 12 2.644 6.519 8.126 4.874 6.48v.001z"
                    />
                  </svg>
                </span>
                <input
                  aria-checked="false"
                  class="yc-checkbox__control"
                  type="checkbox"
                  value=""
                />
                <span
                  class="yc-checkbox__outline"
                />
              </span>
            </label>
          </td>
          <td
            class="yc-table__cell yc-table__cell_edge-padding"
          >
            —
          </td>
          <td
            class="yc-table__cell yc-table__cell_sticky_right yc-table__cell_edge-padding"
            style="width: 28px; right: 0px;"
          />
        </tr>
        <tr
          class="yc-table__row yc-table__row_disabled"
        >
          <td
            class="yc-table__cell yc-table__cell_edge-padding yc-table__checkbox_cell"
          >
            <label
              class="yc-control-label yc-control-label_size_l yc-control-label_disabled yc-checkbox yc-checkbox_size_l yc-checkbox_disabled yc-table__selection-checkbox"
            >
              <span
                class="yc-control-label__indicator yc-checkbox__indicator"
              >
                <span
                  aria-hidden="true"
                  class="yc-checkbox__icon"
                >
                  <svg
                    class="yc-checkbox__icon-svg yc-checkbox__icon-svg_type_tick"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 12 10"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M.49 5.385l1.644-1.644 4.385 4.385L4.874 9.77.49 5.385zm4.384 1.096L10.356 1 12 2.644 6.519 8.126 4.874 6.48v.001z"
                    />
                  </svg>
                </span>
                <input
                  aria-checked="false"
                  class="yc-checkbox__control"
                  disabled=""
                  type="checkbox"
                  value=""
                />
                <span
                  class="yc-checkbox__outline"
                />
              </span>
            </label>
          </td>
          <td
            class="yc-table__cell yc-table__cell_edge-padding"
          >
            —
          </td>
          <td
            class="yc-table__cell yc-table__cell_sticky_right yc-table__cell_edge-padding"
            style="width: 28px; right: 0px;"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Table HOCs tests withTableActions should match snapshot 1`] = `
<div>
  <div
    class="yc-table"
  >
    <table
      class="yc-table__table"
    >
      <thead
        class="yc-table__head"
      >
        <tr
          class="yc-table__row"
        >
          <th
            class="yc-table__cell yc-table__cell_edge-padding"
          >
            name
          </th>
          <th
            class="yc-table__cell yc-table__cell_sticky_right yc-table__cell_edge-padding"
            style="width: 28px; right: 0px;"
          />
        </tr>
      </thead>
      <tbody
        class="yc-table__body"
      >
        <tr
          class="yc-table__row"
        >
          <td
            class="yc-table__cell yc-table__cell_edge-padding"
          >
            —
          </td>
          <td
            class="yc-table__cell yc-table__cell_sticky_right yc-table__cell_edge-padding"
            style="width: 28px; right: 0px;"
          />
        </tr>
        <tr
          class="yc-table__row yc-table__row_disabled"
        >
          <td
            class="yc-table__cell yc-table__cell_edge-padding"
          >
            —
          </td>
          <td
            class="yc-table__cell yc-table__cell_sticky_right yc-table__cell_edge-padding"
            style="width: 28px; right: 0px;"
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
